airbyte-server/src/main/java/io/airbyte/server/ServerApp.java

```diff
@@ -137,36 +134,29 @@ public void start() throws Exception {
   }
 
   private static void assertDatabasesReady(final Configs configs,
-                                           final DatabaseInstance configsDatabaseInstance,
-                                           final DataSource configsDataSource,
-                                           final DatabaseInstance jobsDatabaseInstance,
-                                           final DataSource jobsDataSource)
-      throws InterruptedException {
+                                           final DSLContext configsDslContext,
+                                           final Flyway configsFlyway,
+                                           final DSLContext jobsDslContext,
+                                           final Flyway jobsFlyway)
+      throws DatabaseCheckException {
     LOGGER.info("Checking configs database flyway migration version..");
-    MinimumFlywayMigrationVersionCheck.assertDatabase(configsDatabaseInstance, MinimumFlywayMigrationVersionCheck.DEFAULT_ASSERT_DATABASE_TIMEOUT_MS);
-    final Flyway configsFlyway = FlywayFactory.create(configsDataSource, ServerApp.class.getName(), ConfigsDatabaseMigrator.DB_IDENTIFIER,
-        ConfigsDatabaseMigrator.MIGRATION_FILE_LOCATION);
-    val configsMigrator = new ConfigsDatabaseMigrator(configsDatabaseInstance.getInitialized(), configsFlyway);
-    MinimumFlywayMigrationVersionCheck.assertMigrations(configsMigrator, configs.getConfigsDatabaseMinimumFlywayMigrationVersion(),
-        configs.getConfigsDatabaseInitializationTimeoutMs());
+    DatabaseCheckFactory
+        .createConfigsDatabaseMigrationCheck(configsDslContext, configsFlyway, configs.getConfigsDatabaseMinimumFlywayMigrationVersion(),
```

Comment:
I'm not totally convinced that we should still be checking for this "minimum migration version" as a gate on if the database is ready to use. It seems like we should just wait until the database is _fully_ migrated instead. But I'm fine with keeping this for now to limit the behavioral changes this PR is making.

Reply:
> I'm not totally convinced that we should still be checking for this "minimum migration version" as a gate on if the database is ready to use. It seems like we should just wait until the database is _fully_ migrated instead. But I'm fine with keeping this for now to limit the behavioral changes this PR is making.

@lmossman I think this is a fair thing to discuss/potentially change once we have the new flow in place.  The goal obviously is to move the existing behavior to the new initialization flow to minimize the changes.

For some reason, GH won't let me reply to your comment on the `load()` method in `BootloaderApp`:  originally, I had the initialization in the main method.  The problem with that is that it forces the Cloud wrapped bootloader to duplicate the startup code, instead of allowing the `BootloaderApp` class to do it internally.  Again, this is all throw away code that will be replaced when we move to the application framework.